clang/lib/Tooling/DependencyScanning/DependencyScanningTool.cpp

```diff
@@ -305,6 +305,10 @@ void IncludeTreePPConsumer::finalize(CompilerInstance &CI) {
     if (!addFile(FilePath))
       return;
   }
+  // Add profile files.
+  addFile(CI.getCodeGenOpts().ProfileInstrumentUsePath, true);
```

Comment:
If the file fails to load, the whole compilation will fail with an error, so there's no value continuing, you could return like 
```
if (!addFile(CI.getCodeGenOpts().ProfileInstrumentUsePath, true))
  return
```

Reply:
This ignores file error because some path can totally be empty or not used and I don't want to have the logic to be in Depscanner. I guess it could be confusing if the file is missing later but I don't have big concerns about the file being found earlier but dropped here.

Reply:
In principle if the files are not going to be used then we shouldn't add them as unnecessary dependencies, meaning the driver should not add them to cc1 flags.

In any case, I don't have a strong opinion on whether it should error or not but could you pass the parameter to `addFile()` as `/*IgnoreFileError*/ true` for readability, along with a comment why it's ok for these files to be missing? I actually missed during code-review that the call will intentionally ignore the error.